core/controllers/classifier.py

```diff
@@ -134,42 +87,119 @@ def post(self):
             email_manager.send_job_failure_email(job_id)
             raise self.InternalErrorException(
                 'The current status of the job cannot transition to COMPLETE.')
-        try:
-            classifier_services.store_classifier_data(job_id, classifier_data)
-        except Exception as e:
-            raise self.InternalErrorException(e)
+
+        classifier_data_proto = getattr(
+            payload_proto.job_result,
+            payload_proto.job_result.WhichOneof('classifier_frozen_model'))
+        classifier_services.store_classifier_data(
+            job_id, classifier_data_proto)
 
         # Update status of the training job to 'COMPLETE'.
         classifier_services.mark_training_job_complete(job_id)
 
         return self.render_json({})
 
+    @acl_decorators.open_access
+    def get(self):
+        """Handles GET requests.
+
+        Retrieves the name of the file on GCS storing the trained model
+        parameters and transfers it to the frontend.
+        """
+        exploration_id = self.request.get('exploration_id')
+        state_name = self.request.get('state_name')
 
-class NextJobHandler(base.BaseHandler):
+        try:
+            exp_version = int(self.request.get('exploration_version'))
+            exploration = exp_fetchers.get_exploration_by_id(
+                exploration_id, version=exp_version)
+            interaction_id = exploration.states[state_name].interaction.id
+        except:
+            raise self.InvalidInputException(
+                'Entity for exploration with id %s, version %s and state %s '
+                'not found.' % (
+                    exploration_id, self.request.get('exploration_version'),
+                    state_name))
+
+        if interaction_id not in feconf.INTERACTION_CLASSIFIER_MAPPING:
+            raise self.PageNotFoundException(
+                'No classifier algorithm found for %s interaction' % (
+                    interaction_id))
+
+        algorithm_id = feconf.INTERACTION_CLASSIFIER_MAPPING[
+            interaction_id]['algorithm_id']
+        algorithm_version = feconf.INTERACTION_CLASSIFIER_MAPPING[
+            interaction_id]['algorithm_version']
+
+        state_training_jobs_mapping = (
+            classifier_services.get_state_training_jobs_mapping(
+                exploration_id, exp_version, state_name))
+        if state_training_jobs_mapping is None:
+            raise self.InvalidInputException
```

Comment:
Please always add some error text to the exception raise, ditto below.

Reply:
Done.